### PR TITLE
Fixes #25787 - Make login and taxonomy settings optional

### PR DESCRIPTION
**Pull request by tbrisker**

### CODE REVIEW

**Comment by tbrisker** [comment]
@ekohl - i hope this is what you meant in https://github.com/theforeman/puppet-foreman/pull/690#issuecomment-443428342 ?

File: manifests/init.pp [hunk 1]
```diff
@@ -30,9 +30,9 @@
 #
 # $email_smtp_password::          Password for SMTP server auth, if authentication is enabled
 #
-# $locations_enabled::            Enable locations?
+# $locations_enabled::            (Deprecated) Enable locations?
```
Comment (ekohl) [inline manifests/init.pp]: We typically don't deprecate in this way. I'd prefer to move it to `Advanced parameters` and write that it's useless starting 1.21.

File: manifests/params.pp [hunk 2]
```diff
@@ -9,8 +9,8 @@
   # should foreman manage host provisioning as well
   $unattended     = true
   $unattended_url = undef
-  # Enable users authentication (default user:admin pw:changeme)
-  $authentication = true
+  # Enable users authentication (default user:admin pw:changeme) (Deprecated)
```
Comment (ekohl) [inline manifests/params.pp]: I'd prefer to mention it's unused starting 1.21 so we know when to remove it.

File: manifests/init.pp (hunk 1 above)
Comment (tbrisker) [reply manifests/init.pp]: updated and moved

File: manifests/params.pp (hunk 2 above)
Comment (tbrisker) [reply manifests/params.pp]: updated

File: manifests/init.pp [hunk 3]
```diff
@@ -50,8 +46,12 @@
...
 # $unattended_url::               URL hosts will retrieve templates from during build (normally http as many installers don't support https)
 #
-# $authentication::               Enable user authentication. Initial credentials are set using admin_username
-#                                 and admin_password.
+# $locations_enabled::            Enable locations? (Unused since 1.21)
+#
+# $organizations_enabled::        Enable organizations? (Unused since 1.21)
+#
+# $authentication::               Enable user authentication. (Unused since 1.21)
+#                                 Initial credentials are set using admin_username and admin_password.
```
Comment (ekohl) [inline manifests/init.pp]: Looks like there was also an oversight here: these are now prefixed with `initial_`.

**Comment by ekohl** [comment]
There's a merge conflict now :(

File: manifests/init.pp (hunk 3 above)
Comment (tbrisker) [reply manifests/init.pp]: fixed on the rebase

**Comment by tbrisker** [comment]
@ekohl thanks, rebased on latest master now.

File: manifests/init.pp [hunk 4]
```diff
@@ -71,7 +71,7 @@
 # $ssl::                          Enable and set require_ssl in Foreman settings (note: requires passenger, SSL does not apply to kickstarts)
 #
 # $repo::                         This can be a specific version or nightly
-#
+
```
Comment (mmoll) [inline manifests/init.pp]: this change should be reverted.

**Comment by mmoll** [comment]
@tbrisker one tiny thing :)

File: manifests/init.pp (hunk 4 above)
Comment (tbrisker) [reply manifests/init.pp]: gah, missed that on the rebase. fixed now.

**Comment by mmoll** [comment]
merged, thanks @tbrisker!

**Pull request merged**